browser/js/calculator.js

```diff
@@ -22,229 +28,247 @@
-      : null
-  }
-
-  function addListeners() {
-    document.querySelector( elementSelectors.BUTTON_SECTION )
+  Calculator.prototype.addListeners = function () {
+    //FIXME: ghetto binding
+    var me = this
+    this.dom.querySelector( Calculator.elementSelectors.BUTTON_SECTION )
       .addEventListener( 'click', function( event ) {
-        if ( event.target.matches( elementSelectors.NUMBER ) ) {
-          enterDigit( event.target.dataset.value )
+        if ( event.target.matches( Calculator.elementSelectors.NUMBER ) ) {
+          me.enterDigit( event.target.dataset.value )
         } else if (
-          ( event.target.matches( elementSelectors.OPERATOR )) &&
-          ( !event.target.matches( elementSelectors.EQUALS ))
+          ( event.target.matches( Calculator.elementSelectors.OPERATOR )) &&
+          ( !event.target.matches( Calculator.elementSelectors.EQUALS ))
         ) {
-          setOperator( event.target.dataset.value )
-        } else if ( event.target.matches( elementSelectors.CLEAR ) ) {
-          clearCalculator()
-        } else if ( event.target.matches( elementSelectors.EQUALS ) ) {
-          calculate()
-        } else if ( event.target.matches( elementSelectors.TOGGLE_SIGN ) ) {
-          toggleNegativeValue()
-        } else if ( event.target.matches( elementSelectors.DOT ) ) {
-          addDot()
-        } else if ( event.target.matches( elementSelectors.PERCENT ) ) {
-          convertPercent()
+          me.setOperator( event.target.dataset.value )
+        } else if ( event.target.matches( Calculator.elementSelectors.CLEAR ) ) {
+          me.clearCalculator()
+        } else if ( event.target.matches( Calculator.elementSelectors.EQUALS ) ) {
+          me.calculate()
+        } else if ( event.target.matches( Calculator.elementSelectors.TOGGLE_SIGN ) ) {
+          me.toggleNegativeValue()
+        } else if ( event.target.matches( Calculator.elementSelectors.DOT ) ) {
+          me.addDot()
+        } else if ( event.target.matches( Calculator.elementSelectors.PERCENT ) ) {
+          me.convertPercent()
         }
       })
+      this.dom.onkeydown = function( event ) {
+        me.isNumber( event.key )
+          ? ( me.fakeClickEffect( event.key, Calculator.cssClasses.GREY_BUTTON_PRESSED ),
+            me.enterDigit( event.key ) )
+          : me.isOperator( event.key )
+          ? ( me.fakeClickEffect( event.key, Calculator.cssClasses.ORANGE_BUTTON_PRESSED ),
+            me.setOperator( event.key ) )
+          : me.isEquals( event.key )
+          ? ( me.fakeClickEffect( Calculator.buttons.EQUALS, Calculator.cssClasses.ORANGE_BUTTON_PRESSED ),
+            me.calculate() )
+          : me.isClear( event.key )
+          ? ( me.fakeClickEffect( Calculator.buttons.CLEAR, Calculator.cssClasses.GREY_BUTTON_PRESSED ),
+            me.clearCalculator() )
+          : me.isDot( event.key )
+          ? ( me.fakeClickEffect( Calculator.buttons.DOT, Calculator.cssClasses.GREY_BUTTON_PRESSED ),
```

Comment:
Using parents to do two expressions is confusing and error prone. Use if statements.